docs/testing.md

```diff
@@ -0,0 +1,31 @@
+# Testing
+
+## Tips and Caveats when writing tests
+
+
+### Handling async/promises
+
+When writing tests that rely on asynchronous operations, such as writing to the database, take care to make sure that those operations are resolved before any tests that rely on them run. If you need to create database records in a setup function such as `beforeAll`, you will want to make sure all async/promise operations resolve before subsequent tests run. You can make sure multiple await (promise) operations resolve by using [`Promise.all()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) (which takes an iterable of promises).
+
+Here's how that might look:
+
+```
+let a = await someAsyncFun();
```

Comment:
In this example wouldn't `await someAsyncFun();` wait for the resulting `someAsyncFunc` promise to resolve? This would mean `a` is not a promise but the resolved value of the promise? Should this be `let a = someAsyncFunc()`?

Reply:
Oh yeah, yup. I was going to write a before/after example, and then did neither. Good catch